### ISSUE TITLE
Goverter does not re-use conversion methods if only target or only source is pointer

### ISSUE THREAD

**Issue by pauhull** [jmattheis/goverter#114]
Take the following code snippet:
```go
package example

// goverter:converter
// goverter:useZeroValueOnPointerInconsistency yes
type Converter interface {
	AToB(A) B

	// goverter:map Property | SomeConversionMethod
	CToD(*C) D
}

type A struct {
	Property *C
}

type B struct {
	Property *D
}

type C struct {
	Property string
}

type D struct {
	Property string
}

func SomeConversionMethod(s string) string {
	return s + " converted"
}
```

You would expect goverter to re-use the `CToD` method. However, this is the result of the conversion:

```go
// Code generated by github.com/jmattheis/goverter, DO NOT EDIT.

package generated

import example "example"

type ConverterImpl struct{}

func (c *ConverterImpl) AToB(source example.A) example.B {
	var exampleB example.B
	exampleB.Property = c.pExampleCToPExampleD(source.Property)
	return exampleB
}
func (c *ConverterImpl) CToD(source *example.C) example.D {
	var exampleD example.D
	if source != nil {
		var exampleD2 example.D
		exampleD2.Property = example.SomeConversionMethod((*source).Property)
		exampleD = exampleD2
	}
	return exampleD
}
func (c *ConverterImpl) pExampleCToPExampleD(source *example.C) *example.D {
	var pExampleD *example.D
	if source != nil {
		var exampleD example.D
		exampleD.Property = (*source).Property
		pExampleD = &exampleD
	}
	return pExampleD
}
```

This means that `SomeConversionMethod` would be used when using `CToD` but not `AToB`.

This bug occurs only if either the source or the target is a pointer. If none or both are pointers, the conversion generates as expected. I believe this is related to #105.

**Comment by jmattheis** [comment]
Yeah, it's another edge-case similar to the one mentioned in the second comment. I'll add this case to the overlapping method check. This means goverter will fail because there are overlapping method definitions.

So the fix is define it like this:
```go
// goverter:converter
// goverter:useZeroValueOnPointerInconsistency yes
type Converter interface {
	AToB(A) 
	CToD(*C) D
	// goverter:map Property | SomeConversionMethod
	internalCToD(C) D
}
```
Internally, goverter simplifies types until there are primitive types that can be copied by value. The simplification process of `*T` to `*T` is to remove the pointer type on both source and target. this means there is no inbetween step where goverter checks for `T` to `*T` or `*T` to `T` definitions.

The error message will look like this:
```
Error while creating converter method:
    func (goverter/example.Converter).AToB(goverter/example.A) goverter/example.B

| goverter/example.A
|
|      | *goverter/example.C
|      |
|      |       | goverter/example.C
|      |       |
source.Property*
target.Property*
|      |       |
|      |       | goverter/example.D
|      |
|      | *goverter/example.D
|
| goverter/example.B

Overlapping struct settings found.

Move these field related settings:
    goverter:map Property | SomeConversionMethod

from this method:
    func (goverter/example.Converter).CToD(*goverter/example.C) goverter/example.D

To a method you have to create with the following signature:
    func(goverter/example.C) goverter/example.D

Goverter will not use CToD inside the current conversion method, thus, field related settings would be ignored.
```

**Comment by jmattheis** [comment]
Fixed in [v1.3.0](https://goverter.jmattheis.de/changelog).
